typed-racket-test/succeed/test-struct-info.rkt

```diff
@@ -0,0 +1,40 @@
+#lang racket/base
+(module test racket/base
+  (require (for-syntax racket/base syntax/parse racket/struct-info racket/match))
+
+  (define-syntax (get-constr stx)
+    (syntax-parse stx
+      [(_ sname:id)
+       (define si (extract-struct-info (syntax-local-value #'sname (lambda () #f))))
+       (match-define (list* struct-desc constr rst) si)
+       #`'#,(syntax-e constr)]))
```

Comment:
Why `syntax-e`? Can't you just use `constr` here?

Reply:
I guess you are trying to see the _name_ of the binding that is specified, but I think we shouldn't care about that, and should care about it having the right name.

Reply:
then for `check-equal?`, we would be using its runtime value, which is a procedure